src/libethash/ethash.h

```diff
@@ -203,97 +113,29 @@ void ethash_full_delete(ethash_full_t full);
+ethash_return_value_t ethash_full_compute(
 	ethash_full_t full,
-	ethash_params const* params,
-	ethash_h256_t const* header_hash,
+	ethash_h256_t const header_hash,
 	uint64_t const nonce
 );
 /**
- * Get a pointer to the cache object held by the full client
- *
- * @param full    The full client whose cache to request
- * @return        A pointer to the cache held by the full client or NULL
- *                if there was no cache in the first place
+ * Get a pointer to the full DAG data
  */
-ethash_cache* ethash_full_get_cache(ethash_full_t full);
+void const* ethash_full_dag(ethash_full_t full);
 /**
- * Move the memory ownership of the cache somewhere else
- *
- * @param full     The full client whose cache's memory ownership  to acquire.
- *                 After this function concludes it will no longer have a cache.
- * @return         A pointer to the moved cache or NULL if there was no cache in the first place
+ * Get the size of the DAG data
  */
-ethash_cache* ethash_full_acquire_cache(ethash_full_t full);
-
-void ethash_get_seedhash(ethash_h256_t *seedhash, const uint32_t block_number);
-
-// Returns if hash is less than or equal to difficulty
-static inline int ethash_check_difficulty(
-	ethash_h256_t const* hash,
-	ethash_h256_t const* difficulty
-)
-{
-	// Difficulty is big endian
-	for (int i = 0; i < 32; i++) {
-		if (ethash_h256_get(hash, i) == ethash_h256_get(difficulty, i)) {
-			continue;
-		}
-		return ethash_h256_get(hash, i) < ethash_h256_get(difficulty, i);
-	}
-	return 1;
-}
-
-int ethash_quick_check_difficulty(
-	ethash_h256_t const* header_hash,
-	uint64_t const nonce,
-	ethash_h256_t const* mix_hash,
-	ethash_h256_t const* difficulty
-);
-
+uint64_t ethash_full_dag_size(ethash_full_t full);
 
 /**
- * =========================
- * =	DEPRECATED API	   =
- * =========================
- *
- * Kept for backwards compatibility with whoever still uses it. Please consider
- * switching to the new API (look above)
- */
-void ethash_mkcache(ethash_cache* cache, ethash_params const* params, ethash_h256_t const* seed);
-void ethash_full(
-	ethash_return_value* ret,
-	void const* full_mem,
-	ethash_params const* params,
-	ethash_h256_t const* header_hash,
-	uint64_t const nonce
-);
-void ethash_light(
-	ethash_return_value* ret,
-	ethash_cache const* cache,
-	ethash_params const* params,
-	ethash_h256_t const* header_hash,
-	uint64_t const nonce
-);
-/**
- * Compute the memory data for a full node's memory
- *
- * @param mem         A pointer to an ethash full's memory
- * @param params      The parameters to compute the data with
- * @param cache       A cache object to use in the calculation
- * @return            true if all went fine and false for invalid parameters
+ * Calculate the seedhash for a given block number
  */
-bool ethash_compute_full_data(void* mem, ethash_params const* params, ethash_cache const* cache);
+void ethash_get_seedhash(ethash_h256_t *seedhash, uint64_t const block_number);
```

Comment:
no returns passed as pointers, no need to `const`ify basic type parameters. `_camelCase` naming for parameters as in coding standards.

should be:
```C
ethash_h256_t ethash_get_seedhash(uint64_t _blockBumber);
```